### PR TITLE
Define separate config for bulk carry forward

### LINKED ISSUE #5114: Bulk carry forward does not check for required fields
**Describe the bug**
If a user disables carry forward for a required field and uses the new bulk carry forward feature, an error is thrown because it attempts to save the new records with no value in the required field.

Context: If carry forward is disabled for a required field, 1 record can be carried forward, but the user needs to input a value for the required field in the new record before saving it manually.

**To Reproduce**

1. Create a new CO record
2. Click on the gear icon to access the form meta
3. Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
4. Click on the gear icon next to "Show Carry Forward Button"
5. Uncheck a required field
6. Fill out required fields and save the record
7. Input a value higher than 1 for "Bulk Carry Forward Count"
8. Click "Carry Forward" and see error

**Expected behavior**
The feature should detect that there is no value in required fields and prevent the user from using bulk carry forward.

**Screenshots**

https://github.com/user-attachments/assets/d862c633-2ff9-4116-9744-3d8ecc011cf8

**Crash Report**
[Specify 7 Crash Report - 2024-07-18T16_49_24.484Z.txt](https://github.com/user-attachments/files/16286497/Specify.7.Crash.Report.-.2024-07-18T16_49_24.484Z.txt)

Please, also fill out the following information manually:

- OS: Mac
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: Fitz_NHMA_Oct_2023
- Collection name: NHMA Monitoring
- User Name: petrifiedroadkill

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #5114

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

- Create a new CO record
- Save
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show bulk carry forward count"
- [ ] Verify you cannot Uncheck a required field (set as required in Schema Config)
- Define you bulk carry forward config (might have to reload the page for the carry config to be applied)
- Input a value higher than 2 for "Bulk Carry Forward Count"
- Click Carry Forward 
- [ ] Verify it respected your config

### CODE REVIEW

**Review by sharadsw — APPROVED** [review]

**Review by emenslin — CHANGES_REQUESTED** [review]
### Testing instructions

- Create a new CO record
- Save
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show bulk carry forward count"
- [ ] Verify you cannot Uncheck a required field (set as required in Schema Config)
- Define you bulk carry forward config (might have to reload the page for the carry config to be applied)
- Input a value higher than 2 for "Bulk Carry Forward Count"
- Click Carry Forward 
- [ ] Verify it respected your config 

Can uncheck required fields 

When I unchecked Prep Type trying to carry forward threw an error and no records were created 
Screenshot 2024-07-23 092635
[Specify 7 Crash Report - 2024-07-23T14_26_36.336Z.txt](https://github.com/user-attachments/files/16350428/Specify.7.Crash.Report.-.2024-07-23T14_26_36.336Z.txt)

When just cataloger is unchecked it creates the records anyway 

https://github.com/user-attachments/assets/def9048e-03d2-4fd6-ba59-b6ea6265d7ae

**Comment by emenslin** [comment]
Also forgot to include this in my initial comment but this is really annoying if there's anyway to fix it 

https://github.com/user-attachments/assets/00ff8d0d-866f-417b-b508-b8a3d395bbe5

If you hover over the carry forward gear the tool tip for it blocks the bulk carry forward gear so you can't do anything until it goes away

**Comment by alesan99** [comment]
Currently nested configurations are shared between bulk and normal configs. I originally encountered it with Preparations but I've reproduced it below:

https://github.com/user-attachments/assets/2cd6bbfd-eb3e-4a76-a6ab-5db95c6947de

**Review by emenslin — APPROVED** [review]
### Testing instructions

- Create a new CO record
- Save
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show bulk carry forward count"
- [x] Verify you cannot Uncheck a required field (set as required in Schema Config)
- Define you bulk carry forward config (might have to reload the page for the carry config to be applied)
- Input a value higher than 2 for "Bulk Carry Forward Count"
- Click Carry Forward 
- [x] Verify it respected your config 

Cannot uncheck required fields anymore, works as expected
Screenshot 2024-07-23 121545

**Review by alesan99 — APPROVED** [review]
- Create a new CO record
- Save
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show bulk carry forward count"
- [x] Verify you cannot Uncheck a required field (set as required in Schema Config)
- Define you bulk carry forward config (might have to reload the page for the carry config to be applied)
- Input a value higher than 2 for "Bulk Carry Forward Count"
- Click Carry Forward 
- [x] Verify it respected your config 

My issue is resolved! All settings between bulk and normal carry are fully separate 👍

Everything else is working as expected as well. All changes to the config appear to apply instantly without having to reload too

File: specifyweb/frontend/js_src/lib/components/FormMeta/CarryForward.tsx [hunk 1]
```diff
@@ -367,8 +407,18 @@ function CarryForwardCategory({
                 }
               >
                 <Input.Checkbox
-                  checked={f.includes(carryForward, field.name)}
-                  disabled={isUnique}
+                  checked={
+                    f.includes(carryForward, field.name) ||
+                    (isBulkConfig === true &&
+                      (field.localization.isrequired ||
+                        field.overrides.isRequired))
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/FormMeta/CarryForward.tsx:414]: could you extract this into a variable to not duplicate it twice? it's a pretty complex expression

**Review by maxpatiiuk — APPROVED** [review]

**Comment by specifysoftware** [comment]
This pull request has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/specify-7-9-7-release-announcement/1979/1

**Pull request merged**